onnxruntime/test/fuzzing/src/OnnxPrediction.cpp

```diff
@@ -2,144 +2,125 @@
 // Licensed under the MIT License.
 
 #include "OnnxPrediction.h"
+#include "onnxruntime_session_options_config_keys.h"
+
+#include <vector>
 
 // Uses the onnxruntime to load the model
 // into a session.
 //
-OnnxPrediction::OnnxPrediction(std::wstring& onnx_model_file)
-:
-rawModel{nullptr},
-ptrSession{nullptr},
-session{ env, onnx_model_file.c_str(), emptySessionOption},
-inputNames{session.GetInputCount()},
-outputNames{session.GetOutputCount()}
-{
+OnnxPrediction::OnnxPrediction(std::wstring& onnx_model_file, Ort::Env& env)
+    : raw_model{nullptr},
+      ptr_session{std::make_unique<Ort::Session>(env, onnx_model_file.c_str(), empty_session_option)},
+      input_names(ptr_session->GetInputCount()),
+      output_names(ptr_session->GetOutputCount()) {
   init();
 }
 
 // Uses the onnx to seri
 //
-OnnxPrediction::OnnxPrediction(onnx::ModelProto& onnx_model)
-:
-session{nullptr}
-{
-  rawModel = std::shared_ptr<void>{alloc.Alloc(onnx_model.ByteSizeLong()),   
-  [this](void * ptr)
-      { 
-      this->GetAllocator().Free(ptr); 
-      }
-  };
+OnnxPrediction::OnnxPrediction(onnx::ModelProto& onnx_model, Ort::Env& env) {
+  raw_model = std::shared_ptr<void>(alloc.Alloc(onnx_model.ByteSizeLong()), [this](void* ptr) {
+    this->GetAllocator().Free(ptr);
+  });
 
-  onnx_model.SerializeToArray(rawModel.get(), static_cast<int>(onnx_model.ByteSizeLong()));
+  onnx_model.SerializeToArray(raw_model.get(), static_cast<int>(onnx_model.ByteSizeLong()));
 
-  ptrSession = std::make_unique<Ort::Session>(env, 
-      rawModel.get(), 
-      onnx_model.ByteSizeLong(), 
-      emptySessionOption),
+  ptr_session = std::make_unique<Ort::Session>(env,
+                                               static_cast<void*>(raw_model.get()),
+                                               onnx_model.ByteSizeLong(),
+                                               empty_session_option);
 
-  inputNames.resize(ptrSession->GetInputCount());
-  outputNames.resize(ptrSession->GetOutputCount());
+  // TODO Use reserve instead of resize
+  input_names.resize(ptr_session->GetInputCount());
```

Comment:
nit: Can this TODO be to do the initialization of input_names and output_names in init() instead of doing resize here and in other ctors, and adding the actual values in init()?